packages/amplify-e2e-tests/src/aws-matchers/iamMatcher.ts

```diff
@@ -27,3 +27,49 @@ export const toBeIAMRoleWithArn = async (roleName: string, arn?: string) => {
   };
   return result;
 };
+
+export const toHaveAssumeRolePolicyConditionMatchingIdpId = async (roleName: string, idpId: string) => {
```

Comment:
Nit: I think something like `toHaveValidAssumeRolePolicyCondition` would be easier to understand at a glance.